src/main/java/twofive/utils/Parser.java

```diff
@@ -0,0 +1,150 @@
+package twofive.utils;
+
+import twofive.command.ByeCommand;
+import twofive.command.Command;
+import twofive.command.DeadlineCommand;
+import twofive.command.DeleteCommand;
+import twofive.command.DueDateCommand;
+import twofive.command.EventCommand;
+import twofive.command.FindCommand;
+import twofive.command.ListCommand;
+import twofive.command.MarkCommand;
+import twofive.command.ToDoCommand;
+import twofive.command.UnmarkCommand;
+import twofive.exception.EmptyDateException;
+import twofive.exception.EmptyDeadlineException;
+import twofive.exception.EmptyDescriptionException;
+import twofive.exception.EmptyEndTimeException;
+import twofive.exception.EmptyKeywordException;
+import twofive.exception.EmptyStartTimeException;
+import twofive.exception.EmptyTasknumException;
+import twofive.exception.InvalidCommandException;
+import twofive.exception.MissingArgumentException;
+
+/**
+ * Parses the contents of a given command to obtain the action
+ * intended to be performed by the user.
+ */
+public class Parser {
+    // Ensure valid task number is provided
+    private static int validateTaskNum(String input) throws EmptyTasknumException, NumberFormatException {
+        String[] taskNumSplit = input.split(" ");
+        if (taskNumSplit.length <= 1 || taskNumSplit[1].trim().equals("")) {
+            throw new EmptyTasknumException();
+        } else {
+            int taskNum = Integer.parseInt(taskNumSplit[1]) - 1;
+            return taskNum;
+        }
+    }
+
+    // Ensure a valid description is provided when adding a new task
+    private static String validateDescription(String input, String commandWord) throws EmptyDescriptionException {
```

Comment:
Perhaps you could consider writing javadocs for this method as it appears to do a non-trivial task